.github/workflows/rust.yml

```diff
@@ -26,40 +26,31 @@ env:
   # For slow tests, this can be reduced using `#[property_test(config = "ProptestConfig { .. }")]`.
   PROPTEST_CASES: 10000
   PROPTEST_MAX_GLOBAL_REJECTS: 100000
+  # Disable spot instances in the GitHub merge queue. Otherwise, inherit the spot setting from the
+  # runner config. This environment variable helps us avoid nested strings inside complex actions
+  # expressions.
+  SPOT: ${{ contains(github.ref, 'gh-readonly-queue') && '/spot=false' || '' }}
 
 jobs:
   cargo-fmt:
-    runs-on: ${{ fromJson(github.repository_owner == 'autonomys' &&
-      '"runs-on=${{ github.run_id }}/runner=self-hosted-ubuntu-22.04-x86-64"' || '"ubuntu-22.04"') }}
+    # Format checks do not require significant resources, so we use a free runner.
+    runs-on: ubuntu-22.04
 
     steps:
       - name: Checkout
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
-      - name: Install Protoc
-        uses: arduino/setup-protoc@c65c819552d16ad3c9b72d9dfd5ba5237b9c906b # v3.0.0
-        with:
-          repo-token: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Configure cache
-        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
-
       - name: cargo fmt
         run: cargo fmt --all -- --check
 
   cargo-clippy:
+    name: cargo-clippy (${{ strategy.job-index == 0 && 'Linux' || (strategy.job-index == 1 && 'Windows' || 'macOS') }})
```

Comment:
Convenient, but I'd argue it is helpful to have exact OS name and version. There are no tests for aarch64 Ubuntu/Windows, but it'd be nice to have at least Ubuntu aarc64 here, in which case "Linux" will be duplicated.

The solution to shifting job names for branch protection rules could be to have a "summary" job that depends on all others. Here is an example:
https://github.com/nazar-pc/abundance/blob/b0eb4ede65e73c8e8d5d2b03414e45dd43526d78/.github/workflows/rust.yml#L381-L400

Note that it uses job names as in yaml file and waits for all of them in case of job matrix. The you'll not need to customize the names here at all and it'll be nicer for long-term maintenance.

Reply:
I'd like to go with OS name and maybe architecture, we can add OS version if we ever run multiple OS versions in CI.